maven-plugin/src/main/java/com/marvinformatics/formatter/LineEnding.java

```diff
@@ -21,56 +21,56 @@
  */
 public enum LineEnding {
 
-	AUTO(System.getProperty("line.separator")),
-	KEEP(null),
-	LF("\n"),
-	CRLF("\r\n"),
-	CR("\r"),
-	UNKNOW(null);
+  AUTO(System.getProperty("line.separator")),
+  KEEP(null),
+  LF("\n"),
+  CRLF("\r\n"),
+  CR("\r"),
+  UNKNOW(null);
 
-	private final String chars;
+  private final String chars;
 
-	LineEnding(String chars) {
-		this.chars = chars;
-	}
+  LineEnding(String chars) {
+    this.chars = chars;
+  }
 
-	public String getChars() {
-		return chars;
-	}
+  public String getChars() {
+    return chars;
+  }
 
-	/**
-	 * Returns the most occurring line-ending characters in the file text or null if no line-ending occurs the most.
-	 * 
-	 * @return
-	 */
-	public static LineEnding determineLineEnding(String fileDataString) {
-		int lfCount = 0;
-		int crCount = 0;
-		int crlfCount = 0;
+  /**
+   * Returns the most occurring line-ending characters in the file text or null if no line-ending occurs the most.
+   * 
+   * @return
+   */
+  public static LineEnding determineLineEnding(String fileDataString) {
+    int lfCount = 0;
+    int crCount = 0;
+    int crlfCount = 0;
 
-		for (int i = 0; i < fileDataString.length(); i++) {
-			char c = fileDataString.charAt(i);
-			if (c == '\r') {
-				if ((i + 1) < fileDataString.length()
-						&& fileDataString.charAt(i + 1) == '\n') {
-					crlfCount++;
-					i++;
-				} else {
-					crCount++;
-				}
-			} else if (c == '\n') {
-				lfCount++;
-			}
-		}
+    for (int i = 0; i < fileDataString.length(); i++) {
+      char c = fileDataString.charAt(i);
+      if (c == '\r') {
+        if ((i + 1) < fileDataString.length()
+            && fileDataString.charAt(i + 1) == '\n') {
+          crlfCount++;
+          i++;
```

Comment:
Refactor the code in order to not assign to this loop counter from within the loop body.